### PR TITLE
Screenshot failed steps

### PR DESCRIPTION
**Pull request by koterpillar**

### CODE REVIEW

File: aloe_webdriver/screenshot_failed.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+from aloe import after, before, world
+
+# Pylint cannot infer the attributes on world
+# pylint:disable=no-member
+
+
+FORMAT = 'failed_{feature_file}_{scenario_index}_{scenario_name}{outline_index}'
+
+
+@before.each_example
```
Comment (danni) [inline aloe_webdriver/screenshot_failed.py]: Is this generally useful? Should we make it available as step.outline_index or something?

File: docs/other.rst [hunk 2]
```diff
@@ -74,3 +74,12 @@ CSS Selectors
     import aloe_webdriver.css
 
 .. automodule:: aloe_webdriver.css
+
+Screenshots
+===========
+
```
Comment (danni) [inline docs/other.rst:80]: Mention TAKE_SCREENSHOTS?

Comment (koterpillar) [inline docs/other.rst:80]: That's just for testing, it's not in the actual API.

File: aloe_webdriver/screenshot_failed.py (hunk 1 above)
Comment (koterpillar) [inline aloe_webdriver/screenshot_failed.py]: I generally dislike the fake generated scenario, need to think about that. One approach would be to store all the generated scenarios somewhere and use `.index()` to find out.

Also, no one needed this before, need to wait for more consumers?

File: docs/other.rst (hunk 2 above)
Comment (danni) [inline docs/other.rst:80]: Why not make it API?

File: aloe_webdriver/screenshot_failed.py (hunk 1 above)
Comment (danni) [inline aloe_webdriver/screenshot_failed.py]: I think I used it in the outputter? Worth checking how that works. If it's private API perhaps rename it to `_outline_index` ?

File: docs/other.rst (hunk 2 above)
Comment (koterpillar) [inline docs/other.rst:80]: If you really want this, add that line around the import. On all the projects I'm using this, I haven't had a need to turn it off conditionally; do you think it'll be useful? Especially when nothing else is configured via environment variables?

**Pull request merged**